[Feature] add LynxContext as params of LynxResourceRequest.

nhsprite opened the pull request:
As described, add LynxContext as params of LynxResourceRequest. so that ResourceFetcher providers could be aware of current request lynx instance.

AutoSubmit: true
doc: https://github.com/lynx-family/lynx/issues/1425
issue: #1425

<!--
  Thank you for submitting a pull request!

  We appreciate the time and effort you have invested in making these changes. Please ensure that you provide enough information to allow others to review your pull request.

  Upon submission, your pull request will be automatically assigned with reviewers.

  If you want to learn more about contributing to this project, please visit: https://github.com/lynx-family/lynx/blob/develop/CONTRIBUTING.md.
-->

## Summary

<!-- Can you explain the reasoning behind implementing this change? What problem or issue does this pull request resolve? -->

<!-- It would be helpful if you could provide any relevant context, such as GitHub issues or related discussions. -->

## Checklist

<!--- Check and mark with an "x" -->

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/lynx-family/lynx?pullRequest=2084) <br/>All committers have signed the CLA.

nhsprite commented:
/land

jianliang00 commented:
🛬 Landing process started! >>695.0

nhsprite commented:
/land

jianliang00 commented:
🛬 Landing process started! >>702.0

nhsprite commented:
/land

nhsprite commented:
/land

jianliang00 commented:
🛬 Landing process started! >>721.0

nhsprite commented:
/land

jianliang00 commented:
🛬 Landing process started! >>759.0

nhsprite committed:
[Feature] add LynxContext as params of LynxResourceRequest.

•  Add LynxContext as a parameter to LynxResourceRequest
•  Update implementations on Android, Darwin (iOS/macOS), and Harmony
•  Enables resource fetchers/providers to access the current Lynx instance for a request

doc: https://github.com/lynx-family/lynx/issues/1425

AutoSubmit: true
doc: #1425
issue: #1425

nhsprite commented:
/land

jianliang00 commented:
🛬 Landing process started! >>768.0

nhsprite commented:
/land

jianliang00 commented:
🛬 Landing process started! >>773.0

nhsprite commented:
/land

jianliang00 commented:
🛬 Landing process started! >>774.0

jianliang00 commented:
🛬 Landing process started! >>775.0

nhsprite commented:
/land

jianliang00 commented:
🛬 Landing process started! >>777.0

nhsprite commented:
/land

jianliang00 commented:
🛬 Landing process started! >>778.0
